core/ecschema-editing/src/Merging/SchemaMerger.ts

```diff
@@ -64,52 +64,58 @@ export class SchemaMerger {
   public async merge(targetSchema: Schema, sourceSchema: Schema): Promise<Schema> {
```

Comment:
Doing the merging with a generic merger that has all these specializations and resolvers is actually more complex than just writing the code straight out.  It pushes us to write all kinds of wonky things like the SchemaItemFactory and the resolver to merge individual attributes of each schema item type.

I think the abstraction as a whole makes it harder to understand what is happening which would normally be an OK tradeoff if it was making things a lot simpler.  In this case I think it makes is more complex.

Reply:
I added a new issue for this #6074